src/server/scripts/Northrend/Naxxramas/boss_faerlina.cpp

```diff
@@ -119,20 +130,42 @@ class boss_faerlina : public CreatureScript
                 return;
 
             if (!urand(0, 3))
+            {
                 Talk(SAY_SLAY);
-
+            }
             if (pInstance)
+            {
                 pInstance->SetData(DATA_IMMORTAL_FAIL, 0);
+            }
         }
 
         void JustDied(Unit*  killer) override
         {
             BossAI::JustDied(killer);
             Talk(SAY_DEATH);
+            if (GameObject* go = me->FindNearestGameObject(SPIDER_WEB_ENTRANCE, 100.0f))
+            {
+                go->SetGoState(GO_STATE_ACTIVE);
+            }
         }
 
         void UpdateAI(uint32 diff) override
         {
+            if (!me->IsInCombat() && sayGreet)
+            {
+                for( std::list<uint64>::iterator itr = summons.begin(); itr != summons.end(); ++itr )
+                {
+                    if (pInstance)
+                    {
+                        if (Creature* cr = pInstance->instance->GetCreature(*itr))
+                        {
+                            if (cr->IsInCombat())
+                                DoZoneInCombat();
+                        }
+                    }
+                }
+            }
```

Comment:
same code suggestion as above, as the code is the same. We could also consider re-using this code somehow